app/controllers/concerns/verify_profile_concern.rb

```diff
@@ -0,0 +1,24 @@
+module VerifyProfileConcern
+  private
+
+  def finish_profile_route
+    decorated_user = current_user.decorate
+    return 'profile' unless decorated_user.pending_profile_requires_verification?
+    return 'verify_profile_phone' if decorated_user.needs_profile_phone_verification?
+    return 'verify_profile' if decorated_user.needs_profile_usps_verification?
+    'profile'
+  end
+
+  def finish_profile_path
```

Comment:
thanks for clarifying -- maybe something not as crazy verbose as your example but more verbose would be helpful: `profile_or_verify_profile_path` ?